File: app/models/pbx_user_schedule.rb
class PbxUserSchedule < RestModel
    include ActiveModel::Validations

    property :day_fri
    property :day_mon
    property :day_sat
    property :day_sun
    property :day_thu
    property :day_tue
    property :day_wed
    property :end_at
    property :end_time
    property :id
    property :metric
    property :name
    property :pbx_user_id #Display Only
    property :priority
    property :start_time
    property :start_at
    property :oncall_behavior
    property :forward_behavior

    validates :name, presence: true

    def self.load_from_json(json)
        PbxUserSchedule.from_source(JSON.parse(json)).first
    end

end

File: frontend/app/models/pbx-user-schedule.js
import DS from 'ember-data';
import pbxUserSchedule from '../validations/pbxUserSchedule'

export default DS.Model.extend(pbxUserSchedule, {
  dayFri: DS.attr('boolean', {defaultValue: false}),
  dayMon: DS.attr('boolean', {defaultValue: false}),
  daySat: DS.attr('boolean', {defaultValue: false}),
  daySun: DS.attr('boolean', {defaultValue: false}),
  dayThu: DS.attr('boolean', {defaultValue: false}),
  dayTue: DS.attr('boolean', {defaultValue: false}),
  dayWed: DS.attr('boolean', {defaultValue: false}),
  metric: DS.attr(),
  name: DS.attr(),
  pbxUserId: DS.attr(), //Display Only
  pbxMyPhoneId: DS.attr(),
  priority: DS.attr('number'),
  startTime: DS.attr(),
  endTime: DS.attr(),
  startAt: DS.attr(),
  endAt: DS.attr(),
  oncallBehavior: DS.attr(),
  forwardBehavior: DS.attr(),
  pbxMyPhone: DS.belongsTo('myphone'),
});
